### PR TITLE
TST: add ppc64le to Travis CI matrix

### PR DESCRIPTION
**Pull request by tylerjereddy**
Add `ppc64le` Linux testing to Travis CI. Part of the effort in #12688.

I don't have Travis hooks active on my fork, so this WIP CI testing / dev could get annoying, but I'll start working on this here for now, for convenience.

Related template: https://github.com/ghatwala/travis-jobdeploytest/blob/master/.travis.yml

### CODE REVIEW

**Comment by tylerjereddy** [comment]
@gerrith3 @ghatwala 

It looks to me like this is working already! Maybe you can quickly confirm, but I see an IBM-tagged gcc compiler and `ppc64le` is all over the place for OS name and paths. There's also a mouse-over on the matrix entry in our Travis CI display showing `linux-ppc64le`.

It looks like I don't even have to specify the distribution name as done in the template -- seems that the switch to `xenial` happens automatically anyway from the initial `trusty` designation / default.

For other core devs -- looks like the unit tests "just pass--" this is what we're expecting, right?

**Comment by tylerjereddy** [comment]
I think the problem with approach used by @akhileshbhople at the time of writing is that it doubles our entire Travis CI test matrix because a top-level os: directive is added. I suspect we'll probably just want one or two extra ppc64le jobs to start, unless they're exempt from the queue limit or something, so simply using the os: directive for ppc64 in a single matrix entry may be easiest for now.

File: .travis.yml [hunk 1]
```diff
@@ -71,6 +71,8 @@ matrix:
     - python: 3.6
       env:
        - NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=1
+    - os: linux-ppc64le
+      python: 3.6
```
Comment (mattip) [inline .travis.yml]: Maybe add a dummy `env:` variable to indicate this is ppc, the [test output matrix](https://travis-ci.org/numpy/numpy/builds/477580933) does not have job names so it is difficult to tell what is different about this run. Or is there a different way to label each job in the matrix?

**Comment by ghatwala** [comment]
@tylerjereddy 

> I think the problem with approach used by @akhileshbhople at the time of writing is that it doubles our entire Travis CI test matrix because a top-level os: directive is added. I suspect we'll probably just want one or two extra ppc64le jobs to start, unless they're exempt from the queue limit or something, so simply using the os: directive for ppc64 in a single matrix entry may be easiest for now.
- If maintaining the same count of Travis jobs ( + 2 ppc64le) jobs is the need , we are fine with your PR addition in just a single matrix. Looks good to me!!!

File: .travis.yml (hunk 1 above)
Comment (tylerjereddy) [reply .travis.yml]: You can mouse over the penguin icon to find a ppc label, but good point nonetheless -- I've added a dummy env variable

**Comment by tylerjereddy** [comment]
I've removed the WIP tag -- note that this will use ATLAS as the linear algebra backend for ppc64le Ubuntu. I'll write a short paragraph about eventually switching to wheel-matched openblas for ppc64le in the tracking issue.

**Pull request merged**